lib/configurators/aws/ec2discovery.go

```diff
@@ -14,10 +14,25 @@
 
 package aws
 
-import "fmt"
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	mrand "math/rand"
+	"time"
+)
 
 func EC2DiscoverySSMDocument(proxy string) string {
-	return fmt.Sprintf(ec2DiscoverySSMDocument, proxy)
+	randomBytes := make([]byte, 4)
+	if _, err := crand.Read(randomBytes); err != nil {
+		// on error from crypto rand fallback to less secure math random
+		mathRand := mrand.New(mrand.NewSource(time.Now().UnixNano()))
+		binary.LittleEndian.PutUint32(randomBytes, mathRand.Uint32())
+	}
+	randString := hex.EncodeToString(randomBytes)
+
+	return fmt.Sprintf(ec2DiscoverySSMDocument, randString, proxy, randString)
```

Comment:
Suggestion: Inline the ec2DiscoverySSMDocument string here, or maybe move the const to the method? It's not reused and it's easier to reason about the format.

Reply:
Done, thank you!